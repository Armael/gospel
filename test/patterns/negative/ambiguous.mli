val f : int -> int
(*@ y = f x
    ensures match x with
    | _ | _ when x = 1 -> true
    | _ -> false *)
