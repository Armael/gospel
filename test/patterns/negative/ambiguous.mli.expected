
*******************************
********** Parsed file ********
*******************************
val f : int -> int[@@gospel
                    {| y = f x
    ensures match x with
    | _ | _ when x = 1 -> true
    | _ -> false |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

val f : int -> int
(*@ y = f x
    ensures ...
     *)
File "ambiguous.mli", line 3, characters 12-72:
Error: Ambiguous or-pattern under guard.
